src/core/Core.cpp

```diff
@@ -59,13 +76,37 @@ Core::Core()
             }
             else
             {
-                return MethodResult::SUCCESS;
+                common::Product product;
+                product = getProduct(productID);
+                product.setCurrentQuantity(currentQuantity-quantity);
+                if(calculateDemandIndicator(quantity,product.getLastBuy(),productID)<5 || currentQuantity-1 == minQuantity)
+                {
+                    EmailData data;
+                    data.productName = product.getName();
+                    getEmailDataToServer(data);
+                }
+                std::time_t timer;
+                time(&timer);
+                product.setLastBuy(timer);
+                MethodResult value = updateProduct(productID,product);
+                if(value == MethodResult::SUCCESS) return MethodResult::SUCCESS;

```

Comment:
pamiętaj, żeby dawac do if/elsow {} i umieszczac je w nowej lini. Poprawi to czytelnosc
```
if
{
...
}
else
{
...
}
```